waku/node/peer_manager/peer_manager.nim

```diff
@@ -993,6 +999,24 @@ proc new*(
     # Leave by default 20% of connections for service peers
     maxRelayPeersValue = maxConnections - (maxConnections div 5)
 
+  var maxServicePeersValue = 0
+  var d_low = 4
+
+  if maxServicePeers.isSome():
+    if maxServicePeers.get() > maxConnections - d_low:
+      error "Max number of service peers can't be greater than the max amount of connections minus d_low",
+        maxConnections = maxConnections, maxServicePeers = maxServicePeers.get()
+      raise newException(
+        Defect,
+        "Max number of service peers can't be greater than the max amount of connections minus d_low",
+      )
+    maxServicePeersValue =
+      min(maxServicePeers.get(), maxConnections - maxRelayPeersValue)
+      # relay has higher priority then service peers
+  else:
+    maxServicePeersValue =
+      max(maxConnections - maxRelayPeersValue, maxConnections div 5)
```

Comment:
what's the logic of this line? Why the `maxConnections div 5`?

Reply:
My logic is to reserve 20% for service peers if the user does not provide maxService, as we discuss in nwaku pm we need to change whole logic